src/MUSI6106Exec/MUSI6106Exec.cpp

```diff
@@ -5,46 +5,89 @@
 #include "MUSI6106Config.h"
 
 #include "AudioFileIf.h"
+#include "CombFilterIf.h"
+/**
+ * 
+| _ 3rdparty: (3rd party dependencies)
+| _ sndlib : sndfile library(3rdparty with ugly code and lots of warnings)
+| _ cmake.modules : (cmake scripts)
+| _ inc : global headers
+| _ src : source code
+| _ AudioFileIO : library wrapping sndfile(3rdparty)
+| _ CombFilter : combfilter library(assignment 1)
+| _ inc : internal headers
+| _ MUSI6106Exec : code for executable binary
+
+*/
 
 using std::cout;
 using std::endl;
 
 // local function declarations
 void    showClInfo ();
-
+void errorMsg(char *msg);
 /////////////////////////////////////////////////////////////////////////////////
 // main function
 int main(int argc, char* argv[])
 {
-    std::string             sInputFilePath,                 //!< file paths
-                            sOutputFilePath;
-
-    static const int        kBlockSize = 1024;
-
-    clock_t                 time = 0;
-
-    float                   **ppfAudioData = 0;
-
-    CAudioFileIf            *phAudioFile = 0;
-    std::fstream            hOutputFile;
+    std::string sInputFilePath,                 //!< file paths
+                sOutputFilePath,
+				sOutputWavFilePath,
+				sPath;
+	sPath = ".\\data\\";
+    static const int kBlockSize = 1024;
+
+    clock_t time = 0;
+	int l;
+	int ndelay;
+	char opt = '\0';
+
+    float **ppfAudioData = 0;
+	float **ppfAudioDataOut = 0;
+	CCombFilterIf *pCCombFilter = 0;
+    CAudioFileIf *phAudioFile = 0;
+	CAudioFileIf *phAudioFile2 = 0;
+    std::fstream hOutputFile;
     CAudioFileIf::FileSpec_t stFileSpec;
 
     showClInfo();
 
     //////////////////////////////////////////////////////////////////////////////
     // parse command line arguments
-    if (argc < 2)
-    {
-        cout << "Missing audio input path!";
-        return -1;
-    }
+	if (argc != 4) {
```

Comment:
According to the assignment, here you would need to add a condition where if the User enters 0 command line arguments, you run the tests you have defined. As you have not defined any test methods, I recommend you do so before submission and add the if functionality to run the tests here:
Namely, the FIR delay=period length test, IIR test for increasing magnitude at delaylength=period, multiple block sizes as input, and check for the processing of zero input signals